### DISCUSSION TITLE
Add Node Here / Instantiate Scene Here / Move Here, but in 3D

### DISCUSSION THREAD

**Discussion by passivestar** [godotengine/godot-proposals#8976] [Editor]
I'm surprised I couldn't find a proposal for this, but I found these in 2d, and it would be great to have them in 3d as well:

![image](https://github.com/godotengine/godot-proposals/assets/60579014/882a5243-56c3-4895-9779-a2fb6cd74e5f)

Because there's no way to quickly move a newly created node to a specific place in the scene. The closest thing is Align Transform with View but it's only good for lights and cameras. Scenes can be dragged from the FileSystem dock, but finding them through a popup like in `Ctrl+Shift+A` is a bit faster. And for nodes (i.e if you want to place a particle system) there's currently no solution

~~That popup menu could be called through `Ctrl+Right Click` in 3d viewport~~

Edit: instead of ctrl it can use the same approach as [this PR](https://github.com/godotengine/godot/pull/90165)

Alternatively `Ctrl+A` and `Ctrl+Shift+A` could cast a ray when you call them and add nodes/scenes where it hits. This could be an editor setting to preserve the current behaviour

**Comment by KoBeWi** [comment]
There is a proposal: https://github.com/godotengine/godot-proposals/issues/435
